### ISSUE TITLE
Suggestion: Be able to activate mousemode only on keypress, keyrelease deactivates

### ISSUE THREAD

**Issue by pyriand3r** [jbensmann/xmouseless#2]
Hi,
i am struggling with a behaviour i would like to achive, maybe it is not possible with the actual implementation or i am doing something wrong so this is maybe a feature request, maybe just a configuration question.
I'm under manjaro + i3 and would like to be able to activate xmouseless mouse-mode when i press (and hold) the caps_lock key and disable it when i release the key. 

I have totally disabled the caps_lock key toggling by setting `setxkbmap -option caps:none`

Nevertheless it is not working. When i press and hold caps_lock xmouseless activates for about 1-2 seconds but i can not move the mouse and then it disables itself.

Is it possible to achive this behaviour in the current implementation or is it possible to implement it?

**Comment by jbensmann** [comment]
Hi, I think the problem here is that xmouseless grabs the whole keyboard and it cannot do so while you hold down a key that is bound globally by i3 (or by any other program). So this is not possible with the current implementation, and unfortunately I don't think there is an easy way to fix this.

**Comment by kyoshiro** [comment]
Yes indeed. This would imitate the behavior of the UHK (I like very much). ATM xmouseless runs only when started by the keybinding e.g. from i3, grabs the whole keyboard and exits on special keys like a,s,d,f,space,escape.
To realize the activation on key-press only, xmouseless may have to be auto started and the key binding to activate xmouseless keyboard grab must not overlap the wm's global key bindings or the other way round. On key-release the keyboard grab has to be stopped. Maybe I find some spare time to give it a try.

**Comment by jbensmann** [comment]
Hi @kyoshiro,

yes you might be right, when xmouseless runs in the background it may be possible, would  be nice if you find a solution!
